rfcs/xxxx-custom-fields.md

```diff
@@ -0,0 +1,94 @@
+
+# Examples
+
+Based on all the input we got from the community, we identified a few key use cases we’d like to focus on first:
+
+### Map
+
+A content editor could directly search for a location on a map. Instead of entering latitude and longitude coordinates.
+
+### Videos
+
+Embedding a Youtube video would display a nice preview, which would make it easier to parse the content on a glimpse. Instead of having to click on the link to see what the video is about.
+
+### Color picker
+
+Select any color in any format (HEX, RGB, CMYK) directly in Strapi.
+
+### URL
+
+Fetch the data from a URL to show the favicon of the page, as well as the title. Instead of plain text.
+
+### Ecommerce
+
+A more complex is the one where a field would fetch external data. Products from a BigCommerce store for instance. And directly display them in the Content Manager.
+
+# Motivation
+
+Being and staying the most customizable CMS is our goal. Fields are no exception.
+
+Strapi currently supports 10 fields (excluding the dynamic zones and components). However, the possibilities of field types are endless. To empower the community to solve their use cases, we want to offer full autonomy to the fields needed by enabling new API capabilities on the front-end and back-end sides.
+
+The content edition is at the core of Strapi’s value. Managing entries should be as natural as possible. Sometimes, it doesn’t work as well when you try to shoehorn something that doesn’t feel intuitive. For instance, entering a URL for a video.
+
+At a later stage, we would also like to solve more advanced use cases, or even rethink some fundamental features such as the Components and Dynamic Zones for more customizations.
+
+# Detailed design
+
+![Installation steps](./flow.png)
+
+There are 3 main areas impacted by this feature:
+
+### 1. Strapi Market & Settings
+
+The first step is for a user to install the field. We are still debating on how custom fields should be installed in Strapi. Via a plugin or through a field package.
+
+In any case, the next step is to configure the field if need be. Entering API keys for Google Maps or Shopify, for instance. 
+
+### 2. Content-Type Builder
+
+![content-type builder](./ctb.png)
+
+Once done, the custom field is available in the “custom” tab of the fields picker modal in the Content-Type Builder.
+
+The configuration flow for a custom field should be the same as the native ones. An injection zone allows custom settings to be added in the “Advanced Settings” tab.
+
+Note: the custom field badge is a combination of color and icon provided in the Design System. 
+
+![badges](./badges.png)
+
+Since we are not introducing any new field type, the content type schema won’t change in structure. This also means the custom fields have the same options as the native ones (ex: required, unique, etc), as well as some additional validation rules. 
```

Comment:
Could you explain what you mean by `as well as some additional validation rules` ?

Are you referring to validations in the content type builder in order to save a custom field or adding custom validations for a custom field in the content manager?

Reply:
The latter. For instance, insuring that a URL for a video is actually from Youtube in the case of a video embed. Or validating that a string is a URL, etc. Is it clearer?